packages/client/hmi-client/src/services/data.ts

```diff
@@ -365,88 +321,60 @@ const getRelatedDocuments = async (docid: string, dataset: string | null) => {
 };
 
 const searchXDDArticles = async (term: string, xddSearchParam?: XDDSearchParams) => {
-	const limitResultsCount = xddSearchParam?.perPage ?? XDD_RESULT_DEFAULT_PAGE_SIZE;
+	let searchParams = `term=${term}`;
+	const url = '/xdd/documents?';
 
-	// NOTE when true it disables ranking of results
-	const enablePagination = xddSearchParam?.fullResults ?? false;
+	if (xddSearchParam) {
```

Comment:
I am a little concerned that the `XDDSearchParams` may contain fields that are not actually valid query parameters. Also, the attempt to abstract how some of the search fields are concerted to proper API query parameter names seems to complicate things.

Reply:
Do you mean, there are attributes in the `XDDSearchParams` type that shouldn't be there or they aren't handled by the server yet? Also what does the abstraction complicate?

Reply:
Some of the XDDSearchParams fields such as "type" is not a valid xDD query parameter and is treated differently to filter xDD extractions. Therefore, I don't see a major value in refactoring the various if statements into a switch statement especially when blindly looping through the XDDSearchParams fields.

In addition, some of the XDDSearchParams fields should be combined with others but this knowledge is kind of hidden away and not well documented. By abstracting or refactoring how all fields are treated via the switch statement fallback mechanism, you risk mistreating some of the fields. 

Overall, this part of the code is actually trivial and should not be a major part of the refactoring nor deserve that many discussions! I am OK with your refactoring but was just pointing out a few potential issues and hence the preference of just leaving this part as is!!